cartoframes/viz/map.py

```diff
@@ -378,170 +373,81 @@ def _get_layer_def(layer):
 
 
 def _format_bounds(bounds):
-    if isinstance(bounds, dict):
-        return _dict_bounds(bounds)
-
-    return _list_bounds(bounds)
-
-
-def _list_bounds(bounds):
-    if len(bounds) != 4:
-        raise ValueError('bounds list must have exactly four values in the '
-                         'order: [west, south, east, north]')
-
-    return _dict_bounds({
-        'west': bounds[0],
-        'south': bounds[1],
-        'east': bounds[2],
-        'north': bounds[3]
-    })
-
-
-def _dict_bounds(bounds):
-    if 'west' not in bounds or 'east' not in bounds or \
-       'north' not in bounds or 'south' not in bounds:
-        raise ValueError('bounds must have east, west, north and '
-                         'south properties')
-
-    clamped_bounds = {
-        'west': _clamp(bounds.get('west'), -180, 180),
-        'east': _clamp(bounds.get('east'), -180, 180),
-        'south': _clamp(bounds.get('south'), -90, 90),
-        'north': _clamp(bounds.get('north'), -90, 90)
-    }
-
-    return '[[{west}, {south}], [{east}, {north}]]'.format(**clamped_bounds)
-
-
-def _clamp(value, minimum, maximum):
-    return max(minimum, min(value, maximum))
-
-
-def _get_super_bounds(layers):
-    """"""
-    # TODO: refactor this method:
-    # Compute the bounds in the source class
-    if layers:
-        hosted_layers = [
-            layer for layer in layers
-            if layer.source.type != SourceType.GEOJSON
-        ]
-        local_layers = [
-            layer for layer in layers
-            if layer.source.type == SourceType.GEOJSON
-        ]
+    if bounds is None:
```

Comment:
I am lost here (maybe out of scope?) but it seems ok

Reply:
This is an important change/refactor. Before that, the bounds for the remote sources were computed here using the old context and making the distinction between local/remote sources using the source type. Now the bounds are computed in the Source (for local and remote data) in [[w,s],[e,n]] format. But the user can set the bounds in the Source, Layer or Map, so the format could be also a dict. The `_format_bounds` function checks, clamps and formats the bounds to a unique format (array). The `_compute_bounds` function computes the merge of all the bounds for each layer. The result is a more efficient & cleaner code to get the final bounds for the map.

Reply:
Ok, thanks for the explanation :heart: